HDF5 Not building on Mac after recent  HDF5 update

**Describe the bug**
A clean clone does not build with previously working build steps on a Mac. The bug seems to be isolated to adding the `HDF5_CMAKE_Fortran_FLAGS` in fortran/examples/CMakeLists.txt that was done in one of the recent commits. The error is shown here:
`[ 98%] Linking Fortran executable ../../bin/f90_ex_ph5example
Undefined symbols for architecture arm64:
  "_getenv_", referenced from:
      _MAIN__ in ph5example.f90.o
ld: symbol(s) not found for architecture arm64
collect2: error: ld returned 1 exit status`

**Expected behavior**
With an older commit, or by commenting out line 128 in fortran/examples/CMakeLists.txt, the code builds and can be installed as expected.

**Platform (please complete the following information)**
 - develop branch commit hash 85c1762
 - MacBook Pro with M2 chip running MacOS Ventura 13.5.2
 - LLVM Clang compiler version 16.0.6. However, issue is with Fortran which is GCC 13.2.0 (since llvm does not have one)
 - CMake version 3.27.6 was used for build generation (no issues with this step)
 - CMake options included `CMake Options: -D CMAKE_INSTALL_PREFIX=/Users/ddunning/Documents/playground/Fierro/install/hdf5 -D CMAKE_BUILD_TYPE=Release -D HDF5_BUILD_FORTRAN=ON -D HDF5_ENABLE_PARALLEL=ON -D BUILD_TESTING=OFF`
 - OpenMPI was used, version 4.1.5

**Additional context**
Admittedly I'm not sure what the fix is. I've had issues with Fortran compilers on Macs in the past that have been fixed by various other measures. I am specifically using Apple Command Line Tools 14.3 because there is a known bug with these Fortran compilers with Command Line Tools 15.0 installed. As previously stated, everything is "fixed" by commenting out line 128 in that CMakeLists.txt, so I guess there is a flag that gets added there that Mac doesn't like.

--- a/fortran/examples/ph5example.f90
+++ b/fortran/examples/ph5example.f90
@@ -16,10 +16,10 @@
      PROGRAM DATASET
 
      USE HDF5 ! This module contains all necessary modules
+     USE MPI
 
      IMPLICIT NONE
 
-     INCLUDE 'mpif.h'
      CHARACTER(LEN=10), PARAMETER :: default_fname = "sds.h5"  ! Default name
      CHARACTER(LEN=8), PARAMETER :: dsetname = "IntArray" ! Dataset name
 
@@ -72,10 +72,8 @@
      CALL h5pset_fapl_mpio_f(plist_id, comm, info, error)
 
      !
-     ! Figure out the filename to use.  If your system does not support
-     ! getenv, comment that statement with this,
-     ! filename = ""
-     CALL getenv("HDF5_PARAPREFIX", filename)
+     ! Figure out the filename to use.
+     CALL get_environment_variable("HDF5_PARAPREFIX", filename)
      fnamelen = LEN_TRIM(filename)
      if ( fnamelen == 0 ) then
 	filename = default_fname

